### PR TITLE
Add support for reCAPTCHA to WebhookAgent

### PR DESCRIPTION
**Pull request by knu**
This feature was suggested in [this comment](https://github.com/cantino/huginn/issues/353#issuecomment-156038978).

### CODE REVIEW

File: app/models/agents/webhook_agent.rb [hunk 1]
```diff
@@ -46,10 +49,30 @@ def receive_web_request(params, method, format)
...
+      # check the reCAPTCHA response if required
+      if recaptcha_secret = interpolated['recaptcha_secret'].presence
+        recaptcha_response = params.delete('g-recaptcha-response') or
+          return ["Not Authorized", 401]
+
+        begin
+          response = faraday.post('https://www.google.com/recaptcha/api/siteverify', {
+                                    secret: recaptcha_secret,
+                                    response: recaptcha_response,
+                                    # remoteip: request.env['REMOTE_ADDR'],
```
Comment (cantino) [inline app/models/agents/webhook_agent.rb]: What's this comment for?

File: app/models/agents/webhook_agent.rb [hunk 2]
```diff
@@ -24,6 +26,7 @@ class WebhookAgent < Agent
           For example, "post,get" will enable POST and GET requests. Defaults
           to "post".
         * `response` - The response message to the request. Defaults to 'Event Created'.
+        * `recaptcha_secret` - Setting this to a reCAPTCHA "secret" key makes your agent verify incoming requests with reCAPTCHA.  Don't forget to embed a reCAPTCHA snippet including your "site" key in the originating form(s).
```
Comment (cantino) [inline app/models/agents/webhook_agent.rb]: Cool!  Maybe link to a page on the wiki with an example?

File: app/models/agents/webhook_agent.rb (hunk 1 above)
Comment (knu) [inline app/models/agents/webhook_agent.rb]: REMOTE_ATTR may or may not hold the actual IP address of a client depending on if a reverse proxy / load balancer is configured properly.  I think I can make it an option.

Comment (knu) [inline app/models/agents/webhook_agent.rb]: Added an option `recaptcha_send_remote_addr`.

**Pull request merged**